### PR TITLE
Add exact-real

### PR DESCRIPTION
**Pull request by saraedum**
This adds the libexactreal library and its Python interface pyexactreal.

The recipe closely follows the [e-antic](https://github.com/conda-forge/e-antic-feedstock) recipe.

It contains the scripts necessary to build on Windows. However, these are currently not used since a dependency is not ready on Windows yet.

### CODE REVIEW

File: recipes/exact-real/benchmark.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Author: Julian Rüth <julian.rueth@fsfe.org>
```
Comment (saraedum) [inline recipes/exact-real/benchmark.patch:1]: Upstreamed in https://github.com/flatsurf/exact-real/pull/138.

**Comment by saraedum** [comment]
@conda-forge/help-python-c this is ready for review :)

File: recipes/exact-real/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        - arb
+        - boost-cpp
+        - libeantic >=1.0.2,<2
+        - gmp  # [unix]
+        - mpir  # [win]
+    test:
+      commands:
+        - test -f $PREFIX/include/exact-real/exact-real.hpp  # [unix]
+        - test -f $PREFIX/lib/libexactreal.so                # [linux]
+        - test -f $PREFIX/lib/libexactreal.dylib             # [osx]
```
Comment (xhochy) [inline recipes/exact-real/meta.yaml:43]:
```suggestion
        - test -f $PREFIX/lib/libexactreal${SHLIB_EXT}                # [unix]
```

File: recipes/exact-real/build.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (saraedum) [reply recipes/exact-real/build.sh:1]: Thanks. I was not aware of this.

**Comment by saraedum** [comment]
Thanks a lot @xhochy. I learned something new.

**Pull request merged**